File: 力扣习题/225用队列实现栈/problems.ts
class MyStack {

    q1: number[];
    q2: number[];
    main = false;
    constructor() {
        this.q1 = [];
        this.q2 = [];
    }

    push(x: number): void {
        if (this.main) {
            this.q1.push(x);
        } else {
            this.q2.push(x);
        }
    }

    pop(): number {
        let res = 0;
        if (this.main) {
            while (this.q1.length > 1) {
                this.q2.push(this.q1.shift() as number);
            }
            res = this.q1.shift() as number;
            this.main = false;
        } else {
            while (this.q2.length > 1) {
                this.q1.push(this.q2.shift() as number);
            }
            res = this.q2.shift() as number;
            this.main = true;
        }
        return res;
    }

    top(): number {
        let res = 0;
        if (this.main) {
            while (this.q1.length > 1) {
                this.q2.push(this.q1.shift() as number);
            }
            res = this.q1[0];
            this.q2.push(this.q1.shift() as number);
            this.main = false;
        } else {
            while (this.q2.length > 1) {
                this.q1.push(this.q2.shift() as number);
            }
            res = this.q2[0];
            this.q1.push(this.q2.shift() as number);
            this.main = true;
        }
        return res;
    }

    empty(): boolean {
        if(this.main) {
            return this.q1.length == 0;

        } else {
            return this.q2.length == 0;
        }
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * var obj = new MyStack()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.empty()
 */

File: 力扣习题/225用队列实现栈/problems.py
class MyStack:

    def __init__(self):
        self.q1 = []
        self.q2 = []
        self.main = True

    def push(self, x: int) -> None:
        if self.main:
            self.q1.append(x)
        else:
            self.q2.append(x)


    def pop(self) -> int:
        if self.main:
            while len(self.q1) > 1:
                self.q2.append(self.q1.pop(0))
            res = self.q1.pop(0)
            self.main = False
        else:
            while len(self.q2) > 1:
                self.q1.append(self.q2.pop(0))
            res = self.q2.pop(0)
            self.main = True
        return res


    def top(self) -> int:
        if self.main:
            while len(self.q1) > 1:
                self.q2.append(self.q1.pop(0))
            res = self.q1[0]
            self.q2.append(self.q1.pop(0))
            self.main = False
        else:
            while len(self.q2) > 1:
                self.q1.append(self.q2.pop(0))
            res = self.q2[0]
            self.q1.append(self.q2.pop(0))
            self.main = True
        return res


    def empty(self) -> bool:
        if self.main:
            return len(self.q1) == 0
        else:
            return len(self.q2) == 0



# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()